webpack/assets/javascripts/react_app/redux/actions/users/personalAccessTokens/index.js

```diff
@@ -0,0 +1,96 @@
+import {
+  USERS_PERSONAL_ACCESS_TOKEN_FORM_OPENED,
+  USERS_PERSONAL_ACCESS_TOKEN_FORM_SUCCESS,
+  USERS_PERSONAL_ACCESS_TOKEN_FORM_BUTTON,
+  USERS_PERSONAL_ACCESS_GET_REQUEST,
+  USERS_PERSONAL_ACCESS_GET_SUCCESS,
+  USERS_PERSONAL_ACCESS_GET_FAILURE
+} from '../../../consts';
+import { SubmissionError } from 'redux-form';
+import { ajaxRequestAction } from '../../common';
+
+const fieldErrors = ({ error }) => {
+
+  let errors = error.errors;
+
+  errors._error = errors.base;
+  delete errors.base;
+  return new SubmissionError(errors);
+};
+
+const checkErrors = response => {
+  if (response.ok) {
+    return response;
+  }
+  if (response.status === 422) {
+    // Handle invalid form data
+    return response.json().then(body => {
+      throw fieldErrors(body);
+    });
+  }
+  throw new SubmissionError({
+    _error: [__('Error submitting data: ') + response.statusText]
+  });
+};
+
+export const showForm = personalAccessToken => {
+  return {
+    type: USERS_PERSONAL_ACCESS_TOKEN_FORM_OPENED,
+    payload: {}
+  };
+};
+
+export const submitForm = (userId, name, expiresAt, csrfToken) => {
+  let data = {
+    name,
+    // eslint-disable-next-line camelcase
+    expires_at: expiresAt
+  };
+  const url = `/api/users/${userId}/personal_access_tokens`;
+
+  return dispatch => {
+    // TODO: extract into generic API handler once there are more use cases
+    return fetch(url, {
+      credentials: 'include',
+      method: 'post',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        'X-CSRF-Token': csrfToken
+      },
+      body: JSON.stringify(data)
+    })
+      .then(checkErrors)
+      .then(response => {
+        return response
+          .json()
+          .then(body => ({
+            type: USERS_PERSONAL_ACCESS_TOKEN_FORM_SUCCESS,
+            payload: { body }
+          }))
+          .then(dispatch);
```

Comment:
this dispatch is written wrongly - it should be:
```js
.then(body => 
    dispatch(
      {
         type: USERS_PERSONAL_ACCESS_TOKEN_FORM_SUCCESS,
         payload: { body }
       }
    )
)
```

In general, I've generalized this function for form submtions - have a look at https://github.com/theforeman/foreman/pull/4795/files#diff-35521d7d9795ca3a7f62a33bb3ceef6a